### PR TITLE
Add canary submodule; export policies in main and submodules

### PR DESCRIPTION
**Pull request by tptee**
- Adds a submodule for `serverless-plugin-canary-deployments` permissions. The plugin requires CodeBuild permissions under the hood.
- Outputs group IAM policies in all modules that provide them. This is useful for attaching to an assumed role (see [example in the reference]()).

### CODE REVIEW

File: modules/canary/policy-developer.tf [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+# - Deploy canary functions that use CodeDeploy.
+###############################################################################
+resource "aws_iam_policy" "developer" {
+  name   = "${local.tf_group_developer_name}-canary"
+  path   = "/"
+  policy = "${data.aws_iam_policy_document.developer.json}"
+}
+
+locals {
+  codedeploy_application_name = "sls-${local.service_name}-${local.iam_stage}-Sls${replace(local.service_name, "-", "")}${local.iam_stage}DeploymentApplication"
```
Comment (ryan-roemer) [inline modules/canary/policy-developer.tf]: Can you add a code comment with a link to how the naming / transformation has to work?

Comment (tptee) [reply modules/canary/policy-developer.tf]: Totally! I need to consolidate those comments about the autogenerated name anyway 😄

File: modules/xray/README.md [hunk 2]
```diff
@@ -18,7 +18,8 @@ This module enables [AWS X-ray][] support for [serverless][] framework applicati
 
 ## Overview
 
-This module adds [IAM permissions][xray_iam] to the Lambda execution role created by the Serverless framework as part of its CloudFormation stack so that the role may send trace data to AWS.
+This module adds [IAM permissions][canary_iam] to the Lambda execution role created by the Serverless framework as part of its CloudFormation stack so that the role may create, rollback, and destroy
+canary deploys with CodeDeploy Lambda traffic shifting.
```
Comment (ryan-roemer) [inline modules/xray/README.md]: WARNING -- WRONG FILE!!!

Comment (tptee) [reply modules/xray/README.md]: 🤦‍♂ thanks for catching that!

File: modules/vpc/outputs.tf [hunk 3]
```diff
@@ -1,2 +1,7 @@
-# Outputs
+output "iam_policy_developer_arn" {
```
Comment (ryan-roemer) [inline modules/vpc/outputs.tf]: Did we document in CHANGELOG why we needed these for VPC?

File: modules/canary/policy-developer.tf (hunk 1 above)
Comment (tptee) [reply modules/canary/policy-developer.tf]: Dug a little further and added links to the exact transformation (the plugin calls a builtin Serverless function that removes non-alphanum characters). I changed the `replace` to use an equivalent regex to the one Serverless uses.

File: modules/vpc/outputs.tf (hunk 3 above)
Comment (tptee) [reply modules/vpc/outputs.tf]: Have now!

**Comment by tptee** [comment]
Should be ready for another pass!

File: modules/canary/README.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+  # (Default values)
+  # iam_region          = `*`
+  # iam_partition       = `*`
+  # iam_account_id      = `AWS_CALLER account`
+  # tf_service_name     = `tf-SERVICE_NAME`
+  # sls_service_name    = `sls-SERVICE_NAME`
+  # role_admin_name     = `admin`
+  # role_developer_name = `developer`
+  # role_ci_name        =  ""
```
Comment (ryan-roemer) [inline modules/canary/README.md]: Is this right? Thought it was `ci`

Comment (tptee) [reply modules/canary/README.md]: I'm pretty sure this was a typo...checking against master and fixing!

**Pull request merged**